mcs/build/rules.make

```diff
@@ -14,10 +14,10 @@ _FILTER_OUT = $(foreach x,$(2),$(if $(findstring $(1),$(x)),,$(x)))
 # given $(thisdir), we compute the path to the top directory
 #
 # split_path = $(filter-out .,$(subst /,$(space),$(1)))
-# make_path = ./$(subst $(space),/,$(1))
 # dotdottify = $(patsubst %,..,$(1))
+# make_path = ./$(subst $(space),/,$(1))
 # topdir = $(call make_path,$(call dotdottify,$(call split_path,$(thisdir))))
-topdir := ./$(subst $(space),/,$(patsubst %,..,$(filter-out .,$(subst /,$(space),$(thisdir)))))
+topdir := $(abspath ./$(subst $(space),/,$(patsubst %,..,$(filter-out .,$(subst /,$(space),$(thisdir))))))
```

Comment:
what exactly was malformed? I fought with these last week and abspath causes issues on Windows once you cross the cygwin/native divide